### PR TITLE
add auto xml snippets

### PR DESCRIPTION
**Pull request by ozym**
This looks big but it isn't as bad as that.

The aim is to add a little bit more functionality in the response generation code (the part that builds the `resp/auto.go` file) to build another set of "snippet" XML formatted files (in `resp/auto`) for including in future response work.

The layout is similar to the NRL mechanism, the various "snippets" can be joined together on the fly to build the required sensor/datalogger StationXML response stages.

There is also a directory `resp/files` which can contain hand rolled, or externally derived response files.  I am anticipating adding a `resp/nrl` to store the NRL derived equivalents.

The future direction of development is in using these snippets rather than relying on the auto.go mechanism (although the yaml based configuration will still have some use as per this PR).

The major requirement has been to include the stationxml code for v1.1 as internal. There is some debate as to whether this should actually live in the main directory so that it can be used for other applications.

### CODE REVIEW

File: internal/stationxml/v1.1/decimation.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+func (d *Decimation) SetInputSampleRate(rate float64) {
+	d.InputSampleRate = Frequency{
+		Float: Float{
+			Value: rate,
+		},
+	}
+}
+
+func (d Decimation) Copy() Decimation {
+	return d
```
Comment (bpeng) [inline internal/stationxml/v1.1/decimation.go:21]: This may not make a copy

File: internal/stationxml/v1.1/email.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	Address string `xml:",chardata"`
+}
+
+func (e Email) IsValid() error {
+
+	if !(len(e.Address) > 0) {
+		return fmt.Errorf("empty email")
+	}
+
+	if !(regexp.MustCompile(`^[\w\.\-_]+@[\w\.\-_]+$`).MatchString(e.Address)) {
```
Comment (bpeng) [inline internal/stationxml/v1.1/email.go:18]: the `regexp` could be a `const`

File: internal/stationxml/v1.1/gain.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// FrequencyEnd) in which the SensitivityValue is valid within the number of decibels specified in FrequencyDBVariation.
+type Gain struct {
+	// A scalar that, when applied to the data values, converts the data to different units (e.g. Earth units)
+	Value float64
+	// The frequency (in Hertz) at which the Value is valid.
+	Frequency float64
+}
+
+func (g Gain) Copy() Gain {
+	return g
```
Comment (bpeng) [inline internal/stationxml/v1.1/gain.go:15]: may not be a copy

File: internal/stationxml/v1.1/phone_number.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	if !(len(p.PhoneNumber) > 0) {
+		return fmt.Errorf("empty phone number")
+	}
+
+	if p.AreaCode == 0 {
+		return fmt.Errorf("no area code")
+	}
+
+	if !(regexp.MustCompile(`^[0-9]+-[0-9]+$`).MatchString(p.PhoneNumber)) {
+		return fmt.Errorf("bad phone number: %s", p.PhoneNumber)
```
Comment (bpeng) [inline internal/stationxml/v1.1/phone_number.go:29]: make `regexp` a `const`

File: internal/stationxml/v1.1/root_marshal_test.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		if err != nil {
+			t.Error(err)
+		}
+
+		var s FDSNStationXML
+		err = xml.Unmarshal(x, &s)
+		if err != nil {
+			t.Error(err)
+		}
+
```
Comment (bpeng) [inline internal/stationxml/v1.1/root_marshal_test.go:59]: Do you want to check if `s == test.r`?

File: internal/stationxml/v1.1/data_availability_span.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+	// The maximum time tear (gap or overlap) in seconds between time
+	// series segments in the specified range. </xs:documentation>
+	MaximumTimeTear *float64 `xml:"maximumTimeTear,attr,omitempty"`
+}
+
+func (d DataAvailabilitySpan) IsValid() error {
+
+	if v := d.NumberSegments; v < 1 {
+		return fmt.Errorf("invalid number of segments: %d", v)
```
Comment (bpeng) [inline internal/stationxml/v1.1/data_availability_span.go:31]: can be `d.NumberSegments < 1`?

**Review by bpeng — CHANGES_REQUESTED** [review]
Looks goos and tidy, small changes

File: internal/stationxml/v1.1/decimation.go (hunk 1 above)
Comment (ozym) [reply internal/stationxml/v1.1/decimation.go:21]: It's fine as a copy given that the struct doesn't hold any slices or pointers. (this code will be replaced in a new PR soon).

File: internal/stationxml/v1.1/email.go (hunk 2 above)
Comment (ozym) [reply internal/stationxml/v1.1/email.go:18]: yes, although that just moves it elsewhere in the code. Again, this code will be removed as I will avoid doing validation prior to encoding (will rely on validators after building).

File: internal/stationxml/v1.1/gain.go (hunk 3 above)
Comment (ozym) [reply internal/stationxml/v1.1/gain.go:15]: it's fine as a copy it's not a pointer reference and the struct doesn't contain slice or pointer values.

Comment (ozym) [reply internal/stationxml/v1.1/gain.go:15]: Also will be replaced.

File: internal/stationxml/v1.1/data_availability_span.go (hunk 6 above)
Comment (ozym) [reply internal/stationxml/v1.1/data_availability_span.go:31]: but then I don't have the v as a nice short variable for the error message (just an idiosyncrasy of mine, much like my avoidance of >= and <=).

**Comment by ozym** [comment]
See #1387 for context

**Comment by ozym** [comment]
@bpeng can you please have another look.

**Review by bpeng — APPROVED** [review]
Code looks tidy and tested working

**Pull request merged**